lib/IRGen/IRGenSIL.cpp

```diff
@@ -1163,8 +1163,42 @@ class IRGenSILFunction :
           emitPackCountDebugVariable(Shape);
         }
       });
+
+    if (auto *BGT = llvm::dyn_cast<BoundGenericType>(Ty)) {
```

Comment:
You can also nest non-generic nominals inside other generic nominals, eg

```
struct G<T> {
  struct Inner {}
 }
```

Eg, `G<Int>.Inner` is not a BoundGenericType.

Just use getAnyNominal() instead